packages/cloudflare-pages/src/cloudflare-pages.ts

```diff
@@ -25,14 +32,38 @@ export const defaultOptions: Required<CloudflarePagesOptions> = {
   external: [],
   minify: true,
   emptyOutDir: false,
+  routesJson: true,
 }
 
+type StaticRoutes = { version: number; include: string[]; exclude: string[] }
+
 export const cloudflarePagesPlugin = (options?: CloudflarePagesOptions): Plugin => {
   const virtualEntryId = 'virtual:cloudflare-pages-entry-module'
   const resolvedVirtualEntryId = '\0' + virtualEntryId
+  let config: ResolvedConfig
+  let staticRoutes: StaticRoutes
+  const staticPaths: string[] = []
 
   return {
     name: '@hono/vite-cloudflare-pages',
+    configResolved: async (resolvedConfig) => {
+      config = resolvedConfig
+      const paths = await readdir(config.publicDir, {
```

Comment:
```suggestion
      const paths = await readdir(config.build.outDir, {
```
I tried this plugin on local and found a bug.
JS and CSS files generated by vite are not included in `exclude`, so they can't be loaded.
We need to look at an output directory.

Reply:
Good catch! (in my prototype implementation mentioned above, Vite only generated server-side bundle so `publicDir` was enough)

IIUC, `build.outDir` will be empty on the `configResolved` hook of the initial `vite build` run, so I suppose `readdir` should be called in later hooks like `writeBundle`. Correct me if I'm wrong.

Reply:
@yudai-nkt 
You are right!
Calling `readdir` on the `writeBundle` hook works by excluding `_worker.js`.
<details>
<summary>Codes I tried</summary>

```diff
  return {
    name: "@hono/vite-cloudflare-pages",
    configResolved: async (resolvedConfig) => {
      config = resolvedConfig;
-      const paths = await readdir(config.publicDir, {
-        withFileTypes: true
-      });
-      paths.forEach((p) => {
-        if (p.isDirectory()) {
-          staticPaths.push(`/${p.name}/*`);
-        } else if (p.name !== "_worker.js") {
-          staticPaths.push(`/${p.name}`);
-        }
      });
    },
    resolveId(id) {
      if (id === virtualEntryId) {
        return resolvedVirtualEntryId;
      }
    },
    async load(id) {
      if (id === resolvedVirtualEntryId) {
        return await getEntryContent({
          entry: options?.entry ? Array.isArray(options.entry) ? options.entry : [options.entry] : [...defaultOptions.entry],
          staticPaths
        });
      }
    },
    writeBundle: async () => {
+      const paths = await readdir(config.build.outDir, {
+        withFileTypes: true
+      });
+      paths.forEach((p) => {
+        if (p.isDirectory()) {
+          staticPaths.push(`/${p.name}/*`);
+        } else if (p.name !== "_worker.js") {
+          staticPaths.push(`/${p.name}`);
+        }
+      });
      staticRoutes = {
        version: 1,
        include: ["/*"],
        exclude: staticPaths
      };
```

</details>

However, HonoX generate `.vite/manifest.json` in out dir.
If it's static build, shouldn't it be output?
I think this needs to be fixed in HonoX.

Reply:
As you said, you should specify `config.build.outDir` if you built the JS or CSS files before build server-side with this plugin. But, if you did not build some front files before executing this, the `config.build.outDir` does not exist, and it should be a different directory, e.g., `public` So, I've created the `serveStaticDir` option:

https://github.com/honojs/vite-plugins/pull/121/commits/3d09e1ca0be359e5608fd33cacda4773f37821b2

The default value is `config.build.outDir`, but the user can specify it themselves. This is good. What do you think?

Reply:
> IIUC, `build.outDir` will be empty on the `configResolved` hook of the initial `vite build` run, so I suppose `readdir` should be called in later hooks like `writeBundle.` Correct me if I'm wrong.

Indeed, it will be empty. But, this plugin for server-side. If we use it with client-side files, e.g., JS or CSS, we should build it in a separate build phase like the below:

```console
vite build --mode client && vite build"
```

So, if we build client things first, `config.build.outDir` exists, and we don't have to write `readdir` in `writeBundle`. And if we call `readdir` in `writeBundle`, the test will fail because the `staticPaths` are empty when `getEntryContent` is executed.

Reply:
It appears that the css imported from 3d09e1ca0be359e5608fd33cacda4773f37821b2 is no longer being loaded.
I'm trying with `vite-plugin` branch of the following repository:
https://github.com/ryuapp/honox-tailwind-template/tree/vite-plugin
```json
"now"
{"version":1,"include":["/*"],"exclude":["/.vite/*","/humans.txt","/static/*"]}

"ideal"
{"version":1,"include":["/*"],"exclude":["/.vite/*","/humans.txt","/static/*","/style-BMTRAcdn.css"]}
```

---

> As you said, you should specify config.build.outDir if you built the JS or CSS files before build server-side with this plugin. But, if you did not build some front files before executing this, the config.build.outDir does not exist, and it should be a different directory, e.g., public So, I've created the serveStaticDir option:
> 
> https://github.com/honojs/vite-plugins/commit/3d09e1ca0be359e5608fd33cacda4773f37821b2
> 
> The default value is config.build.outDir, but the user can specify it themselves. This is good. What do you think?

I'm sorry I didn't explain it well.
What I have a problem with is just `.vite/manifest.json` in an output directory.
However, I had misunderstood `manifest.json`; thus that was not an issue to discuss here...

Regarding the option, I don't think it' necessary.
This is because creating `_routes.json` from config.build.outDir is simpler and easier to understand.
If we want to handle `_routes.json` freely, I think that if we have `_routes.json` in public, we should give priority to that.

Reply:
@ryuapp 

Thank you for your explanation! Understood well.

I've added the change: https://github.com/honojs/vite-plugins/pull/121/commits/84b90c498402aaf233526b12e3f2305d2d2ee95e

* Move `readdir` into `writeBundle`.
* Remove `serveStaticDir` and `routesJson` options.
* Use `config.build.outDir`.
* Don't write `serveStatic` code on `_worker.js`.

The point is that it does not add `serveStatic` on `_worker.js` and should generate `_routes.json`. The `serverStatic` is not necessary if it has `_routes.json`, and it will be simpler rather than if both are set.

What do you think of this?

Reply:
@yusukebe 
Great! This PR works well in my project :)
I agree with all your opinions. I have no more questions. 
Thank you very much for your great work!